facebook/detectors/rce/cisco_smi/src/test/java/com/google/tsunami/plugins/detectors/rce/cisco_smi/CiscoSMIDetectorTest.java

```diff
@@ -0,0 +1,161 @@
+    when(socket.getOutputStream()).thenReturn(out);
+    when(socket.getInputStream()).thenReturn(new ByteArrayInputStream(in));
+    return out;
+  }
+
+  @Test
+  public void detect_whenCiscoSMIVulnerable_reportsVuln() throws Exception {
+    ByteArrayOutputStream out = configureMockSocket(new byte[]{
+        0x00, 0x00, 0x00, 0x04, 0x00, 0x00, 0x00, 0x00,
+        0x00, 0x00, 0x00, 0x03, 0x00, 0x00, 0x00, 0x08,
+        0x00, 0x00, 0x00, 0x01, 0x00, 0x00, 0x00, 0x00,
+    });
+    assertThat(detector
+            .detect(localTargetInfo(), ImmutableList.of(smartInstallService()))
+            .getDetectionReportsList()
+    ).containsExactly(
+        DetectionReport.newBuilder()
+            .setTargetInfo(localTargetInfo())
+            .setNetworkService(smartInstallService())
+            .setDetectionTimestamp(Timestamps.fromMillis(fakeUtcClock.millis()))
+            .setDetectionStatus(DetectionStatus.VULNERABILITY_VERIFIED)
+            .setVulnerability(
+                Vulnerability.newBuilder()
+                    .setMainId(
+                        VulnerabilityId.newBuilder()
+                            .setPublisher("Cisco")
+                            .setValue("cisco-sa-20170214-smi"))
+                    .setSeverity(Severity.HIGH)
+                    .setTitle("Cisco Smart Install Protocol Misuse")
+                    .setDescription("Cisco Smart Install feature should not be exposed as it enables attackers to perform administrative tasks on the device or remotely execute code"))
+            .build());
+    assertArrayEquals(new byte[]{
+        0x00, 0x00, 0x00, 0x01, 0x00, 0x00, 0x00, 0x01,
+        0x00, 0x00, 0x00, 0x04, 0x00, 0x00, 0x00, 0x08,
+        0x00, 0x00, 0x00, 0x01, 0x00, 0x00, 0x00, 0x00,
+    }, out.toByteArray());
+  }
+
+  @Test
+  public void detect_whenCiscoSMINotVulnerable_doesNotReportsVuln() throws Exception {
+    configureMockSocket(new byte[]{});
+    assertThat(detector
+            .detect(localTargetInfo(), ImmutableList.of(smartInstallService()))
+            .getDetectionReportsList()
+    ).isEmpty();
+  }
+
+  @Test
+  public void detect_whenCiscoSMIUnreachable_doesNotReportVuln() throws Exception {
+    when(socketFactoryMock.createSocket(anyString(), anyInt())).thenThrow(java.net.SocketException.class);
+    assertThat(detector
+            .detect(localTargetInfo(), ImmutableList.of(smartInstallService()))
+            .getDetectionReportsList()
+    ).isEmpty();
+  }
+
+  @Test
+  public void detect_whenEmptyNetworkService_generatesEmptyDetectionReports() {
+    assertThat(detector
+            .detect(localTargetInfo(), ImmutableList.of())
+            .getDetectionReportsList())
+        .isEmpty();
+  }
+
+    private static TargetInfo localTargetInfo() {
+        return TargetInfo.newBuilder().addNetworkEndpoints(forHostname("localhost")).build();
+    }
+
+  private final static NetworkService smartInstallService() {
```

Comment:
code style (Google Java Style [section 4.8.7](https://google.github.io/styleguide/javaguide.html#s4.8.7-modifiers)): modifier order should be `private static final`.